src/horizontal-pod-autoscaler.ts

```diff
@@ -0,0 +1,866 @@
+        duration: Duration.minutes(1),
+      },
+      {
+        replicas: Replicas.percent(200),
+        duration: Duration.minutes(1),
+      },
+    ],
+  };
+  private readonly _defaultScaleDown = {
+    strategy: ScalingStrategy.MAX_CHANGE,
+    stabilizationWindow: Duration.minutes(5),
+    policies: [
+      {
+        replicas: Replicas.absolute(this.minReplicas ?? 1),
+        duration: Duration.minutes(5),
+      },
+    ],
+  };
+  private readonly _defaultScalingPolicy = {
+    duration: Duration.seconds(15),
+  };
+
+
+  constructor(scope: Construct, id: string, props: HorizontalPodAutoscalerProps) {
+    super(scope, id);
+    this.apiObject = new k8s.KubeHorizontalPodAutoscalerV2(this, 'Resource', {
+      metadata: props.metadata,
+      spec: Lazy.any({ produce: () => this._toKube() }),
+    });
+
+    if (props.target.kind === 'DaemonSet') {
+      throw new Error('HorizontalPodAutoscaler cannot be used with workloads that do not scale, such as a DaemonSet.');
+    }
+    const hasResourceConstraints = props.target.containers.some((c) => this._hasRequestsOrLimits(c));
+    if (!hasResourceConstraints && !props.metrics) {
+      throw new Error('Every container in the HorizontalPodAutoscaler target must have CPU or memory resources defined');
+    }
+    if (props?.minReplicas && props.minReplicas > props.maxReplicas) {
+      throw new Error("'minReplicas' must be less than or equal to maxReplicas in order for HorizontalPodAutoscaler to scale.");
+    }
+    if (props?.scaleUp?.stabilizationWindow && !this._isStabilizationWindowWithinRange(props.scaleUp.stabilizationWindow)) {
+      throw new Error("'scaleUp.stabilizationWindow' must be more than 0 seconds and no longer than 1 hour.");
+    }
+    if (props?.scaleDown?.stabilizationWindow && !this._isStabilizationWindowWithinRange(props.scaleDown.stabilizationWindow)) {
+      throw new Error("'scaleDown.stabilizationWindow' must be more than 0 seconds and no longer than 1 hour.");
+    }
+    if (props?.scaleUp?.policies?.length) {
+      this._validateScalingPolicies(ScalingDirection.UP, props.scaleUp.policies);
+      this._setDefaultsOnScalingPolicies(props.scaleUp.policies);
+    }
+    if (props?.scaleDown?.policies?.length) {
+      this._validateScalingPolicies(ScalingDirection.DOWN, props.scaleDown.policies);
+      this._setDefaultsOnScalingPolicies(props.scaleDown.policies);
+    }
+
+    this.target = props.target;
+    this.maxReplicas = props.maxReplicas;
+    this.minReplicas = props.minReplicas ?? 1;
+    this.metrics = props.metrics ?? undefined;
+    this.scaleUp = {
+      ...this._defaultScaleUp,
+      ...props.scaleUp,
+    };
+    this.scaleDown = {
+      ...this._defaultScaleDown,
+      ...props?.scaleDown,
+    };
+  }
+
+  /**
+   * Validate a list of scaling policies.
+   * @internal
+   */
+  private _validateScalingPolicies(direction: ScalingDirection, policies: ScalingPolicy[]) {
+    return policies.map((p) => {
```

Comment:
Why `return` on a validation function? Its confusing because it gives the impression the return value is being used somewhere, but isn't. 

```suggestion
    policies.forEach((p) => {
```